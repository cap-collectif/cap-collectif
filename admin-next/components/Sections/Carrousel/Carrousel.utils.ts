import { SectionIdCarrouselQuery$data } from '@relay/SectionIdCarrouselQuery.graphql'
import { CarrouselElementType } from '@relay/SectionIdCarrouselQuery.graphql'

export type FormValues = {
  position: number
  enabled: boolean
  isLegendEnabledOnImage: boolean
  preventReSubmit?: boolean
  carrouselElements: Array<{
    id?: string
    title: string
    description: string
    buttonLabel: string
    redirectLink: string
    image: { id: string; url: string }
    type: CarrouselElementType
    isDisplayed: boolean
    defaultIsOpen?: boolean
  }>
}

export type PrefillEntity = {
  readonly description: string | null
  readonly label: string
  readonly media: {
    readonly type: string
    readonly url: string | null
  } | null
  readonly url: string | null
  readonly value: string
}

export const isValid = (value: FormValues) => {
  if (value.preventReSubmit || !value.position || !Number.isInteger(value.position) || value.position < 1) return false
  if (
    value.carrouselElements.every(element => {
      return (
        element.title && ((element.buttonLabel && element.redirectLink) || element.type !== 'CUSTOM') && element.image
      )
    })
  )
    return true
  return false
}

export const getSelectLabel = (type: CarrouselElementType) => {
  switch (type) {
    case 'EVENT':
      return 'section.select_event'
    case 'PROJECT':
      return 'select-a-participatory-project'
    case 'THEME':
      return 'section.select_theme'
    case 'ARTICLE':
    default:
      return 'section.select_article'
  }
}

export const getCardLabel = (type: CarrouselElementType) => {
  switch (type) {
    case 'EVENT':
      return 'type-event'
    case 'PROJECT':
      return 'global.project'
    case 'THEME':
      return 'global.theme'
    case 'ARTICLE':
      return 'global.post'
    case 'CUSTOM':
    default:
      return 'customized'
  }
}

export const getInitialValues = (
  carrouselConfiguration: SectionIdCarrouselQuery$data['carrouselConfiguration'],
  newItemId?: string | null,
) => {
  return {
    ...carrouselConfiguration,
    carrouselElements:
      carrouselConfiguration.carrouselElements?.edges
        ?.map(({ node }) => ({
          ...node,
          defaultIsOpen: node.id === newItemId || undefined,
        }))
        .sort((item1, item2) => (item1.position > item2.position ? 1 : -1)) || [],
    preventReSubmit: newItemId !== undefined,
  }
}

export const MAX_TITLE_LENGTH = 50
export const MAX_DESC_LENGTH = 165
export const MAX_LABEL_LENGTH = 20
