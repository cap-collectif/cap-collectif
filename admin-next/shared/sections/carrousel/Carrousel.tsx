import { Box, BoxProps, ButtonQuickAction, CapUIIcon, CapUIIconSize, Flex } from '@cap-collectif/ui'
import React, { useCallback, useEffect, useRef, useState } from 'react'
import { FC } from 'react'
import { pxToRem } from '@shared/utils/pxToRem'
import { useIntl } from 'react-intl'
import CarrouselItem, { BIG_SCREEN_BAND_HEIGHT, Item, RATIO, SMALL_SCREEN_BAND_HEIGHT } from './CarrouselItem'
import { graphql, useLazyLoadQuery } from 'react-relay'
import { CarrouselQuery } from '@relay/CarrouselQuery.graphql'
import WhatsNew from './WhatsNew'

export type Mode = 'inside' | 'below'

const duration = 3500

// No pause button yet, so no auto animation
// Set this to true to enable auto animation if needed
const AUTO_ANIMATE = false

const CarrouselFullView: FC<{ mode?: Mode; items: Item[] } & BoxProps> = ({ mode = 'below', items, ...rest }) => {
  const intl = useIntl()
  const timer = useRef(null)
  const [currentIndex, setCurrentIndex] = useState(1)
  const [disabled, setIsDisabled] = useState(false)
  const [transitionEnabled, setTransitionEnabled] = useState(true)
  const length = items.length
  const [counter, setCounter] = useState(0)
  const counterValid = AUTO_ANIMATE && counter > -1

  useEffect(() => {
    if (currentIndex === 1 || currentIndex === length) {
      setTransitionEnabled(true)
    }
  }, [currentIndex, length])

  const next = useCallback(() => {
    if (!disabled) {
      setCurrentIndex(prevState => prevState + 1)
      setIsDisabled(true)
    }
  }, [disabled])

  const prev = () => {
    if (!disabled) {
      setCurrentIndex(prevState => prevState - 1)
      setIsDisabled(true)
    }
  }

  const handleTransitionEnd = () => {
    setIsDisabled(false)
    if (currentIndex === 0) {
      setTransitionEnabled(false)
      setCurrentIndex(length)
    } else if (currentIndex === length + 1) {
      setTransitionEnabled(false)
      setCurrentIndex(1)
    }
  }

  useEffect(() => {
    timer.current =
      counterValid &&
      setInterval(() => {
        next()
        setCounter(t => t + 1)
      }, duration)
    return () => clearInterval(timer.current)
  }, [counterValid, next])

  return (
    <Box
      as="section"
      id="carrousel"
      aria-roledescription={intl.formatMessage({ id: 'global.carrousel' })}
      aria-label={intl.formatMessage({ id: 'menu.news' })}
      width="100%"
      overflow="hidden"
      position="relative"
      onMouseEnter={() => setCounter(-1)}
      onMouseLeave={() => setCounter(0)}
      backgroundColor="white"
      {...rest}
    >
      <Box mb={[pxToRem(mode === 'below' ? 60 : 76), 0]}>
        {length > 1 ? (
          <Flex
            position="absolute"
            top={['', mode === 'below' ? `calc(${RATIO} - ${pxToRem(64)})` : 'unset']}
            bottom={[
              4,
              mode === 'below' ? 'unset' : pxToRem(SMALL_SCREEN_BAND_HEIGHT + 16),
              mode === 'below' ? 'unset' : pxToRem(BIG_SCREEN_BAND_HEIGHT + 16),
            ]}
            zIndex={1}
            gap={4}
            left={[`calc(50% - ${pxToRem(52)})`, 'unset', 'unset', `calc(50vw + ${pxToRem(512)})`]}
            right={['', 8, 8, 'unset']}
          >
            <ButtonQuickAction
              icon={CapUIIcon.ArrowLeft}
              aria-controls="carrousel-items"
              label={intl.formatMessage({ id: 'carrousel.prev_slide' })}
              onClick={prev}
              variantColor="hierarchy"
              size={CapUIIconSize.Md}
            />
            <ButtonQuickAction
              icon={CapUIIcon.ArrowRight}
              aria-controls="carrousel-items"
              label={intl.formatMessage({ id: 'carrousel.next_slide' })}
              onClick={next}
              variantColor="hierarchy"
              size={CapUIIconSize.Md}
            />
          </Flex>
        ) : null}

        <Flex
          id="carrousel-items"
          aria-live="off"
          width={`${(items.length + 2) * 100}%`}
          onTransitionEnd={handleTransitionEnd}
          style={{
            transform: `translate(-${(100 * currentIndex) / (items.length + 2)}%)`,
            transition: transitionEnabled ? 'transform 1s' : null,
          }}
        >
          <CarrouselItem slide={items[length - 1]} mode={mode} aria-hidden tabIndex={-1} />
          {items.map((slide, index) => (
            <CarrouselItem
              key={index}
              slide={slide}
              mode={mode}
              aria-label={intl.formatMessage({ id: 'carrousel.position' }, { position: index + 1, total: length })}
              aria-hidden={currentIndex - 1 !== index}
              tabIndex={currentIndex - 1 !== index ? -1 : undefined}
            />
          ))}
          <CarrouselItem slide={items[0]} mode={mode} aria-hidden tabIndex={-1} />
        </Flex>
      </Box>
    </Box>
  )
}

export const QUERY = graphql`
  query CarrouselQuery($type: String = "carrousel") {
    carrouselConfiguration(type: $type) {
      enabled
      title
      isLegendEnabledOnImage
      carrouselElements {
        edges {
          node {
            title
            position
            description
            isDisplayed
            buttonLabel
            type
            redirectLink
            image {
              id
              url
            }
            startAt
            endAt
          }
        }
      }
    }
  }
`

type SectionType = 'carrousel' | 'carrouselHighlighted'

const Carrousel: FC<{ type: SectionType; alt: boolean }> = ({ type = 'carrousel' }) => {
  const data = useLazyLoadQuery<CarrouselQuery>(QUERY, { type })
  if (!data || !data?.carrouselConfiguration?.enabled) return null

  const { carrouselConfiguration } = data
  const { isLegendEnabledOnImage, carrouselElements, title } = carrouselConfiguration
  const items = carrouselElements.edges
    .map(({ node }) => node)
    .filter(item => item.isDisplayed)
    .sort((item1, item2) => (item1.position > item2.position ? 1 : -1))

  const isOdd = !!document?.querySelector('.section--custom:nth-of-type(odd)#whats-new')

  return type === 'carrouselHighlighted' ? (
    <WhatsNew items={items} title={title} backgroundColor={isOdd ? 'GRAY' : 'WHITE'} />
  ) : (
    <CarrouselFullView mode={isLegendEnabledOnImage ? 'inside' : 'below'} items={items} />
  )
}

export default Carrousel
