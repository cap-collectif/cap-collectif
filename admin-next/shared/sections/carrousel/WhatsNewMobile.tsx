import { Box, CapUIIcon, Flex, Text } from '@cap-collectif/ui'
import React, { FC, useRef } from 'react'
import Slider from 'react-slick'
import { pxToRem } from '@shared/utils/pxToRem'
import CarrouselButton from './CarrouselButton'
import { useIntl } from 'react-intl'
import Image from '@shared/ui/Image'
import { Item } from './CarrouselItem'

export const SECTION_TITLE_MAX_LENGTH = 11
const ITEM_TITLE_MAX_LENGTH = 45

const WhatsNewMobile: FC<{ items: Item[] }> = ({ items }) => {
  const intl = useIntl()
  let sliderRef = useRef<null | { slickNext: () => void; slickPrev: () => void }>(null)
  const next = () => {
    sliderRef.current.slickNext()
  }
  const prev = () => {
    sliderRef.current.slickPrev()
  }

  return (
    <Box
      as="section"
      id="whats-new"
      width="100%"
      overflow="hidden"
      position="relative"
      margin="auto"
      pb={4}
      aria-roledescription={intl.formatMessage({ id: 'global.carrousel' })}
      aria-label={intl.formatMessage({ id: 'menu.news' })}
      sx={{
        '.slick-slide': {
          px: 4,
        },
      }}
    >
      {/* <Heading as="h3" mb={4} px={4}>
        {'Ã€ la une'.slice(0, SECTION_TITLE_MAX_LENGTH)}
      </Heading> */}
      <Slider
        ref={slider => {
          sliderRef.current = slider
        }}
        {...{
          infinite: true,
          slidesToScroll: 1,
          slidesToShow: 1,
        }}
      >
        {items.map((item, index) => (
          <Box
            as="a"
            height="100%"
            href={item.redirectLink}
            key={index}
            borderRadius="accordion"
            overflow="hidden"
            bg="neutral-gray.150"
          >
            <Image src={item.image.url} alt="" loading="eager" height={pxToRem(180)} width="100%" />
            <Box p={4}>
              {item.type ? (
                <Text as="div" fontSize={4} mb={1}>
                  {item.type}
                </Text>
              ) : null}
              <Text fontSize={4} fontWeight={600} truncate={ITEM_TITLE_MAX_LENGTH}>
                {item.title}
              </Text>
            </Box>
          </Box>
        ))}
      </Slider>
      {items.length > 1 ? (
        <Flex zIndex={1} gap={4} justifyContent="center" mt={4}>
          <CarrouselButton
            variantColor="neutral-gray"
            icon={CapUIIcon.ArrowLeft}
            label={intl.formatMessage({ id: 'carrousel.prev_slide' })}
            onClick={prev}
          />
          <CarrouselButton
            variantColor="neutral-gray"
            icon={CapUIIcon.ArrowRight}
            label={intl.formatMessage({ id: 'carrousel.next_slide' })}
            onClick={next}
          />
        </Flex>
      ) : null}
    </Box>
  )
}

export default WhatsNewMobile
