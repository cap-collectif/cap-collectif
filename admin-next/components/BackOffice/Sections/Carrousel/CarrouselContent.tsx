import { FC } from 'react'
import { useFieldArray, useFormContext } from 'react-hook-form'
import CarrouselEmptyList from './CarrouselEmptyList'
import { Box, DragnDrop } from '@cap-collectif/ui'
import { CarrouselAddItemButton } from './CarrouselAddItemButton'
import CarrouselItem from './CarrouselItem'
import { FormValues, SectionType } from './Carrousel.utils'
import { useIntl } from 'react-intl'

export const CarrouselContent: FC<{ onDelete: (id: string) => void; title?: string; type: SectionType }> = ({
  onDelete,
  title,
  type,
}) => {
  const { watch, control } = useFormContext<FormValues>()
  const intl = useIntl()

  const { fields, move, prepend } = useFieldArray<FormValues, 'carrouselElements'>({
    control,
    name: 'carrouselElements',
  })
  const watchFieldArray = watch('carrouselElements')
  const controlledFields = fields.map((field, index) => {
    return {
      ...field,
      ...watchFieldArray[index],
    }
  })

  const onDragEnd = (result: { destination: { index: number }; source: { index: number } }) => {
    if (!result.destination || result.destination.index === result.source.index) {
      return
    }
    move(result.source.index, result.destination.index)
  }

  if (!controlledFields.length)
    return (
      <CarrouselEmptyList
        prepend={prepend}
        title={title ? `"${title}"` : intl.formatMessage({ id: 'global.carrousel' })}
        type={type}
      />
    )

  return (
    <Box>
      <CarrouselAddItemButton prepend={prepend} cannotAddMoreFields={controlledFields.length >= 8} type={type} />
      {controlledFields.length ? (
        // @ts-ignore
        <DragnDrop onDragEnd={onDragEnd} backgroundColor="red">
          <DragnDrop.List droppableId="carrouselElements">
            {controlledFields.map((element, index) => (
              // @ts-expect-error MAJ DS Props
              <DragnDrop.Item draggableId={element.id} index={index} key={element.id}>
                <CarrouselItem
                  fieldBaseName={`carrouselElements.${index}`}
                  onDelete={() => onDelete(element.id)}
                  type={type}
                />
              </DragnDrop.Item>
            ))}
          </DragnDrop.List>
        </DragnDrop>
      ) : null}
    </Box>
  )

  return
}

export default CarrouselContent
