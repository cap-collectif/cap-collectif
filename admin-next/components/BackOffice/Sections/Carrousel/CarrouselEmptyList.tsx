import { Box, Flex, Heading, Text } from '@cap-collectif/ui'
import { FC } from 'react'
import { useIntl } from 'react-intl'
import { CarrouselEmptyIllustration } from './CarrouselEmptyIllustration'
import { pxToRem } from '@shared/utils/pxToRem'
import { CarrouselAddItemButton } from './CarrouselAddItemButton'
import { UseFieldArrayPrepend } from 'react-hook-form'
import { FormValues, SectionType } from './Carrousel.utils'

export const CarrouselEmptyList: FC<{
  prepend: UseFieldArrayPrepend<FormValues, 'carrouselElements'>
  title?: string
  type: SectionType
}> = ({ prepend, title, type }) => {
  const intl = useIntl()
  return (
    <Flex spacing={8}>
      <Box>
        <Heading as="h3" mb={2}>
          {intl.formatMessage({ id: 'section.carousel_param' }, { title })}
        </Heading>
        <Text mb={4} lineHeight="base">
          {intl.formatMessage({ id: 'section.carousel_helptext' }, { title })}
        </Text>
        <CarrouselAddItemButton prepend={prepend} type={type} />
      </Box>
      <CarrouselEmptyIllustration width={pxToRem(300)} flex="none" />
    </Flex>
  )
}

export default CarrouselEmptyList
