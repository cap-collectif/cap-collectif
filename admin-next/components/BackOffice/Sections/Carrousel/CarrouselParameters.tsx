import { FieldInput, FormControl } from '@cap-collectif/form'
import { Flex, FormLabel, Heading, Link } from '@cap-collectif/ui'
import { FC } from 'react'
import { useFormContext } from 'react-hook-form'
import { useIntl } from 'react-intl'
import { SectionType } from './Carrousel.utils'

export const SECTION_TITLE_MAX_LENGTH = 20

export const CarrouselParameters: FC<{ type: SectionType }> = ({ type }) => {
  const intl = useIntl()
  const { control } = useFormContext()
  return (
    <Flex direction="column" spacing={6} width="30%">
      <Flex p={6} direction="column" spacing={6} backgroundColor="white" borderRadius="accordion" position="relative">
        <Flex color="blue.500" position="absolute" right={6} top={6}>
          <Link
            href="/"
            target="_blank"
            ml={1}
            fontWeight={600}
            sx={{
              textDecoration: 'none !important',
            }}
          >
            {intl.formatMessage({ id: 'global.preview' })}
          </Link>
        </Flex>
        <Heading as="h4" color="blue.800" fontWeight="600">
          {intl.formatMessage({ id: 'global-customization' })}
        </Heading>
        <div>
          {type === 'carrouselHighlighted' ? (
            <FormControl name="title" control={control} isRequired>
              <FormLabel htmlFor="title" label={intl.formatMessage({ id: 'global.title' })} />
              <FieldInput
                name="title"
                control={control}
                type="text"
                placeholder={intl.formatMessage({
                  id: 'admin.label.highlighted',
                })}
                maxLength={SECTION_TITLE_MAX_LENGTH}
              />
            </FormControl>
          ) : null}
          <FormControl name="position" control={control} isRequired>
            <FormLabel label={intl.formatMessage({ id: 'section-admin-display-order' })} />
            <FieldInput
              type="number"
              name="position"
              control={control}
              min={1}
              step="1"
              rules={{
                validate: value =>
                  (Number.isInteger(value) && value > 0) || intl.formatMessage({ id: 'must-be-positive-integer' }),
              }}
            />
          </FormControl>
          <FormControl
            name="enabled"
            control={control}
            sx={{ label: { justifyContent: type === 'carrousel' ? 'space-between' : 'left' } }}
          >
            <FieldInput id="enabled" type="switch" name="enabled" control={control} direction="row-reverse">
              {intl.formatMessage({ id: 'section.activate' })}
            </FieldInput>
          </FormControl>
          {type === 'carrousel' ? (
            <FormControl
              name="isLegendEnabledOnImage"
              control={control}
              sx={{ label: { justifyContent: 'space-between', span: { flex: '1' } } }}
            >
              <FieldInput
                id="isLegendEnabledOnImage"
                type="switch"
                name="isLegendEnabledOnImage"
                control={control}
                direction="row-reverse"
              >
                {intl.formatMessage({ id: 'section.display_legend_on_picture' })}
              </FieldInput>
            </FormControl>
          ) : null}
        </div>
      </Flex>
    </Flex>
  )
}

export default CarrouselParameters
