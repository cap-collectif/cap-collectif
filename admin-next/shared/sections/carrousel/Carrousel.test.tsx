/* eslint-env jest */
import { render } from '@testing-library/react'
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils'
import { RelaySuspensFragmentTest, MockProviders } from 'tests/testUtils'
import Carrousel from './Carrousel'
import WhatsNewMobile from './WhatsNewMobile'

const item = {
  title: 'Item1',
  position: 0,
  description: 'Lorem ipsum',
  isDisplayed: true,
  buttonLabel: 'Voir +',
  type: 'EVENT',
  redirectLink: '/event/mon-event',
  image: {
    id: 'img1',
    url: '/media/img1.png',
  },
  startAt: null,
  endAt: null,
}

describe('<Carrousel />', () => {
  let environment
  let TestComponent
  const defaultMockResolvers = {
    CarrouselConfiguration: () => ({
      id: 'carrouselId1',
      title: 'Title',
      isLegendEnabledOnImage: true,
      carrouselElements: {
        edges: [
          {
            node: item,
          },
        ],
      },
    }),
  }

  beforeEach(() => {
    environment = createMockEnvironment()
    environment.mock.queueOperationResolver(operation => MockPayloadGenerator.generate(operation, defaultMockResolvers))

    TestComponent = props => (
      <RelaySuspensFragmentTest environment={environment}>
        <Carrousel {...props} />
      </RelaySuspensFragmentTest>
    )
  })
  it('renders correctly with a carrousel', () => {
    const { asFragment } = render(<TestComponent type="carrousel" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders correctly with a carrouselHighlighted', () => {
    const { asFragment } = render(<TestComponent type="carrouselHighlighted" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders correctly with a carrouselHighlighted on mobile', () => {
    const { asFragment } = render(
      <MockProviders>
        <WhatsNewMobile title="TitleMobile" items={[item]} />
      </MockProviders>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
