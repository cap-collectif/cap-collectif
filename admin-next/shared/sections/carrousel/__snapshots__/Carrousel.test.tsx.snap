// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carrousel /> renders correctly with a carrousel 1`] = `
[
  <div
    className="sc-bcXGCL eEZeZH toasts-container toasts-container--top"
  />,
  <section
    aria-label="menu.news"
    aria-roledescription="global.carrousel"
    className="sc-bcXGCL jPGxXS"
    id="carrousel"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bcXGCL lfSKRX"
    >
      <div
        aria-live="off"
        className="sc-bcXGCL fRtILG"
        id="carrousel-items"
        onTransitionEnd={[Function]}
        style={
          {
            "transform": "translate(-33.333333333333336%)",
            "transition": "transform 1s",
          }
        }
      >
        <div
          aria-hidden={true}
          aria-roledescription="carrousel.slide"
          className="sc-bcXGCL gFjRDL"
          role="group"
          tabIndex={-1}
        >
          <div
            className="sc-bcXGCL kTxBHB"
          >
            <img
              alt=""
              className="sc-bcXGCL icjWQy"
              loading="eager"
              sizes="(max-width: 320px) 320px, (max-width: 640px) 640px, (max-width: 960px) 960px, (max-width: 1280px) 1280px, (max-width: 2560px) 2560px,"
              srcSet="
        http://localhost/cdn-cgi/image/width=320,format=auto//media/img1.png   320w,
         http://localhost/cdn-cgi/image/width=640,format=auto//media/img1.png   640w,
         http://localhost/cdn-cgi/image/width=960,format=auto//media/img1.png   960w,
         http://localhost/cdn-cgi/image/width=1280,format=auto//media/img1.png 1280w,
         http://localhost/cdn-cgi/image/width=2560,format=auto//media/img1.png 2560w,
         http://localhost/cdn-cgi/image/dpr=2,format=auto//media/img1.png 2x,
         http://localhost/cdn-cgi/image/dpr=3,format=auto//media/img1.png 3x,
         "
              style={
                {
                  "objectFit": "cover",
                }
              }
            />
          </div>
          <div
            className="sc-bcXGCL jNzoRB"
          >
            <div
              className="sc-bcXGCL kpuzHJ"
            >
              <div
                className="sc-bcXGCL gjsubU"
              >
                <h3
                  className="sc-bcXGCL gSuuRt cap-heading"
                >
                  Item1
                </h3>
                <p
                  className="sc-bcXGCL dxUxdV"
                >
                  Lorem ipsum
                </p>
              </div>
              <a
                className="sc-bcXGCL UEWLU"
                href="/event/mon-event"
                tabIndex={-1}
              >
                Voir +
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={false}
          aria-label="carrousel.position {"position":1,"total":1}"
          aria-roledescription="carrousel.slide"
          className="sc-bcXGCL gFjRDL"
          role="group"
        >
          <div
            className="sc-bcXGCL kTxBHB"
          >
            <img
              alt=""
              className="sc-bcXGCL icjWQy"
              loading="eager"
              sizes="(max-width: 320px) 320px, (max-width: 640px) 640px, (max-width: 960px) 960px, (max-width: 1280px) 1280px, (max-width: 2560px) 2560px,"
              srcSet="
        http://localhost/cdn-cgi/image/width=320,format=auto//media/img1.png   320w,
         http://localhost/cdn-cgi/image/width=640,format=auto//media/img1.png   640w,
         http://localhost/cdn-cgi/image/width=960,format=auto//media/img1.png   960w,
         http://localhost/cdn-cgi/image/width=1280,format=auto//media/img1.png 1280w,
         http://localhost/cdn-cgi/image/width=2560,format=auto//media/img1.png 2560w,
         http://localhost/cdn-cgi/image/dpr=2,format=auto//media/img1.png 2x,
         http://localhost/cdn-cgi/image/dpr=3,format=auto//media/img1.png 3x,
         "
              style={
                {
                  "objectFit": "cover",
                }
              }
            />
          </div>
          <div
            className="sc-bcXGCL jNzoRB"
          >
            <div
              className="sc-bcXGCL kpuzHJ"
            >
              <div
                className="sc-bcXGCL gjsubU"
              >
                <h3
                  className="sc-bcXGCL gSuuRt cap-heading"
                >
                  Item1
                </h3>
                <p
                  className="sc-bcXGCL dxUxdV"
                >
                  Lorem ipsum
                </p>
              </div>
              <a
                className="sc-bcXGCL UEWLU"
                href="/event/mon-event"
              >
                Voir +
              </a>
            </div>
          </div>
        </div>
        <div
          aria-hidden={true}
          aria-roledescription="carrousel.slide"
          className="sc-bcXGCL gFjRDL"
          role="group"
          tabIndex={-1}
        >
          <div
            className="sc-bcXGCL kTxBHB"
          >
            <img
              alt=""
              className="sc-bcXGCL icjWQy"
              loading="eager"
              sizes="(max-width: 320px) 320px, (max-width: 640px) 640px, (max-width: 960px) 960px, (max-width: 1280px) 1280px, (max-width: 2560px) 2560px,"
              srcSet="
        http://localhost/cdn-cgi/image/width=320,format=auto//media/img1.png   320w,
         http://localhost/cdn-cgi/image/width=640,format=auto//media/img1.png   640w,
         http://localhost/cdn-cgi/image/width=960,format=auto//media/img1.png   960w,
         http://localhost/cdn-cgi/image/width=1280,format=auto//media/img1.png 1280w,
         http://localhost/cdn-cgi/image/width=2560,format=auto//media/img1.png 2560w,
         http://localhost/cdn-cgi/image/dpr=2,format=auto//media/img1.png 2x,
         http://localhost/cdn-cgi/image/dpr=3,format=auto//media/img1.png 3x,
         "
              style={
                {
                  "objectFit": "cover",
                }
              }
            />
          </div>
          <div
            className="sc-bcXGCL jNzoRB"
          >
            <div
              className="sc-bcXGCL kpuzHJ"
            >
              <div
                className="sc-bcXGCL gjsubU"
              >
                <h3
                  className="sc-bcXGCL gSuuRt cap-heading"
                >
                  Item1
                </h3>
                <p
                  className="sc-bcXGCL dxUxdV"
                >
                  Lorem ipsum
                </p>
              </div>
              <a
                className="sc-bcXGCL UEWLU"
                href="/event/mon-event"
                tabIndex={-1}
              >
                Voir +
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`<Carrousel /> renders correctly with a carrouselHighlighted 1`] = `
[
  <div
    className="sc-bcXGCL eEZeZH toasts-container toasts-container--top"
  />,
  <div
    aria-label="menu.news"
    aria-roledescription="global.carrousel"
    className="sc-bcXGCL gYleZF"
  >
    <div
      className="sc-bcXGCL ckIgbN"
    >
      <div
        className="sc-bcXGCL jJHnhg"
      >
        <div
          className="sc-bcXGCL fJEreX"
        >
          <div
            className="sc-bcXGCL eZYuXx"
          >
            <h3
              className="sc-bcXGCL jGwkpw cap-heading"
            >
              Title
            </h3>
          </div>
          <div
            aria-live="off"
            className="sc-bcXGCL kizpMN"
            id="whats-new-items"
          >
            <div
              aria-label="carrousel.position {"position":1,"total":1}"
              aria-roledescription="carrousel.slide"
              className="sc-bcXGCL hYIRwa"
              id="whats-new-item-0"
              onClick={[Function]}
              role="group"
            >
              <div
                className="sc-bcXGCL ewzazk"
              >
                <p
                  className="sc-bcXGCL cvCRpc cap-text"
                >
                  type-event
                </p>
              </div>
              <h4
                className="sc-bcXGCL cHdXqd cap-heading"
              >
                Item1
              </h4>
              <p
                className="sc-bcXGCL bnBsPp cap-text"
                title="Lorem ipsum"
              >
                Lorem ipsum
              </p>
              <a
                className="sc-bcXGCL fmHybK cap-link"
                href="/event/mon-event"
              >
                learn.more
                <span
                  className="sr-only"
                >
                  global.on
                   
                  type-event
                   
                  Item1
                </span>
              </a>
            </div>
          </div>
        </div>
        <div
          className="sc-bcXGCL iGOMlb"
        >
          <img
            alt=""
            className="sc-bcXGCL hJtzCJ"
            loading="eager"
            sizes="(max-width: 320px) 320px, (max-width: 640px) 640px, (max-width: 960px) 960px, (max-width: 1280px) 960px, (max-width: 2560px) 960px,"
            srcSet="
        http://localhost/cdn-cgi/image/width=320,format=auto//media/img1.png   320w,
         http://localhost/cdn-cgi/image/width=640,format=auto//media/img1.png   640w,
         http://localhost/cdn-cgi/image/width=960,format=auto//media/img1.png   960w,
         http://localhost/cdn-cgi/image/width=1280,format=auto//media/img1.png 1280w,
         http://localhost/cdn-cgi/image/width=2560,format=auto//media/img1.png 2560w,
         http://localhost/cdn-cgi/image/dpr=2,format=auto//media/img1.png 2x,
         http://localhost/cdn-cgi/image/dpr=3,format=auto//media/img1.png 3x,
         "
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<Carrousel /> renders correctly with a carrouselHighlighted on mobile 1`] = `
[
  <div
    className="sc-bcXGCL eEZeZH toasts-container toasts-container--top"
  />,
  <div
    aria-label="menu.news"
    aria-roledescription="global.carrousel"
    className="sc-bcXGCL xUnqz"
  >
    <h3
      className="sc-bcXGCL eCwHQX cap-heading"
    >
      TitleMobile
    </h3>
    <div
      className="slick-slider slick-initialized"
    >
      <div
        className="slick-list"
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            {
              "WebkitTransform": "translate3d(0px, 0px, 0px)",
              "WebkitTransition": "",
              "msTransform": "translateX(0px)",
              "opacity": 1,
              "transform": "translate3d(0px, 0px, 0px)",
              "transition": "",
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <a
                className="sc-bcXGCL hEmCby"
                href="/event/mon-event"
                style={
                  {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt=""
                  className="sc-bcXGCL kAPKok"
                  loading="eager"
                  sizes="(max-width: 320px) 320px, (max-width: 640px) 640px, (max-width: 960px) 960px, (max-width: 1280px) 1280px, (max-width: 2560px) 2560px,"
                  srcSet="
        http://localhost/cdn-cgi/image/width=320,format=auto//media/img1.png   320w,
         http://localhost/cdn-cgi/image/width=640,format=auto//media/img1.png   640w,
         http://localhost/cdn-cgi/image/width=960,format=auto//media/img1.png   960w,
         http://localhost/cdn-cgi/image/width=1280,format=auto//media/img1.png 1280w,
         http://localhost/cdn-cgi/image/width=2560,format=auto//media/img1.png 2560w,
         http://localhost/cdn-cgi/image/dpr=2,format=auto//media/img1.png 2x,
         http://localhost/cdn-cgi/image/dpr=3,format=auto//media/img1.png 3x,
         "
                  style={
                    {
                      "objectFit": "cover",
                    }
                  }
                />
                <div
                  className="sc-bcXGCL jensdJ"
                >
                  <div
                    className="sc-bcXGCL kyCDBf"
                  >
                    <div
                      className="sc-bcXGCL hQmHcG cap-text"
                    >
                      type-event
                    </div>
                  </div>
                  <p
                    className="sc-bcXGCL lddfrQ cap-text"
                    title="Item1"
                  >
                    Item1
                  </p>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
